### DISCUSSION TITLE
Hiding node segment when there's a bun.lockb file in root

### DISCUSSION THREAD

**Discussion by mkvlrn-cm42** [JanDeDobbeleer/oh-my-posh#5425] [Troubleshoot]
How do I make the [bun](https://ohmyposh.dev/docs/segments/cli/bun) segment "solo", as the [node](https://ohmyposh.dev/docs/segments/languages/node) segment will also show up if there's a `package.json` file or maybe a `node_modules` directory?

Bun's segment is tied to the `bun.lockb` file it seems, so I guess it's about making the node segment also tied to it, but hiding it, correct? If so, how do I do it?

My current segments:

```json
"segments": [
  {
    "foreground": "#6c287e",
    "properties": {},
    "style": "powerline",
    "template": " {{ .Full }} ",
    "type": "dotnet"
  },
  {
    "foreground": "#8ED1F7",
    "properties": {},
    "style": "powerline",
    "template": " {{ .Full }} ",
    "type": "go"
  },
  {
    "foreground": "#6CA35E",
    "properties": {},
    "style": "powerline",
    "template": " {{ .Full }} ",
    "type": "node"
  },
  {
    "foreground": "#3C82F6",
    "properties": {},
    "style": "powerline",
    "template": "🥟 {{ .Full }} ",
    "type": "bun"
  }
]
```

Showing up like this in a fresh `bun init`ed directory:
![image](https://github.com/user-attachments/assets/f8886807-aeb7-41b0-9489-ab383bdf5d1e)

**Comment by JanDeDobbeleer** [comment]
@mkvlrn you can adjust the `extensions` property to include all the files you want (or limit to the ones you want). It's a string array.

**Reply by mkvlrn-cm42** [reply]
Since node and bun share a lot of files and the deciding factor being the `bun.lockb` file, I couldn't find a way to make the `extensions` property do what I wanted.

A search (that I should have done prior to opening this discussion, but hey) for "glob" and "oh-my-posh" pushed me in the right direction.

```json
{
  "foreground": "#6CA35E",
  "properties": {},
  "style": "powerline",
  "template": "{{ if glob \"bun.lockb\" }}{{ else }}󰎙 {{ .Full }} {{ end }}",
  "type": "node"
}
```

Thanks!

**Reply by JanDeDobbeleer** [reply]
@mkvlrn good solution. Indeed, extensions doesn't yet allow exclusions. Could be a useful addition.
